tests/foreman/cli/test_remoteexecution.py

```diff
@@ -1123,3 +1123,76 @@ def test_positive_run_job_on_host_registered_to_pull_provider(
             }
         )
         assert_job_invocation_result(invocation_command['id'], rhel_contenthost.hostname)
+
+    @pytest.mark.tier3
+    @pytest.mark.upgrade
+    @pytest.mark.no_containers
+    @pytest.mark.rhel_ver_list([8])
+    def test_positive_run_pull_job_on_offline_host(
+        self,
+        module_org,
+        module_target_sat,
+        smart_proxy_location,
+        module_ak_with_cv,
+        module_capsule_configured_mqtt,
+        rhel_contenthost,
+    ):
+        """Run pull-mqtt job against offline host
+
+        :id: c4914b78-6414-4a13-87b1-5b5cf01702a0
+
+        :expectedresults: Job is resumed when host comes back online
+
+        :CaseImportance: Critical
+
+        :parametrized: yes
+        """
+        client_repo = ohsnap.dogfood_repository(
+            settings.repos.ohsnap_repo_host,
+            product='client',
+            repo='client',
+            release='Client',
+            os_release=rhel_contenthost.os_version.major,
+        )
+        # Update module_capsule_configured_mqtt to include module_org/smart_proxy_location
+        module_target_sat.cli.Capsule.update(
+            {
+                'name': module_capsule_configured_mqtt.hostname,
+                'organization-ids': module_org.id,
+                'location-ids': smart_proxy_location.id,
+            }
+        )
+        result = rhel_contenthost.register(
+            module_org,
+            smart_proxy_location,
+            module_ak_with_cv.name,
+            target=module_capsule_configured_mqtt,
+            satellite=module_target_sat,
+            setup_remote_execution_pull=True,
+            repo=client_repo.baseurl,
+        )
+
+        assert result.status == 0, f'Failed to register host: {result.stderr}'
+        # check mqtt client is running
+        result = rhel_contenthost.execute('systemctl status yggdrasild')
+        assert result.status == 0, f'Failed to start yggdrasil on client: {result.stderr}'
+        # stop the client on host
+        result = rhel_contenthost.execute('systemctl stop yggdrasild')
+        assert result.status == 0, f'Failed to stop yggdrasil on client: {result.stderr}'
+        # run script provider rex command
+        invocation_command = make_job_invocation(
+            {
+                'job-template': 'Run Command - Script Default',
+                'inputs': 'command=ls',
+                'search-query': f"name ~ {rhel_contenthost.hostname}",
+                'async': True,
+            }
+        )
+        # assert pending
+        assert_job_invocation_status(invocation_command['id'], rhel_contenthost.hostname, 'running')
```

Comment:
You're definitely the REX expert here, but I'm assuming `pending` and `running` are the same thing in this instance?

Reply:
ah yes, the returned status is "running", but the job actually waits until the client picks it up